int32[] data_raw
int32[] data_label